charts/redpanda/chart_template_test.go

```diff
@@ -118,6 +123,36 @@ func TestTemplate(t *testing.T) {
 	}
 }
 
+func AssertVolumeMountsVerification(t *testing.T, manifests []byte) {
+	objs, err := kube.DecodeYAML(manifests, redpanda.Scheme)
+	require.NoError(t, err)
+
+	for _, obj := range objs {
+		sts, ok := obj.(*v1.StatefulSet)
```

Comment:
I think we'll be doing ourselves a lot of favors if we make this work for more than just `StatefulSets`. If you think that's too much work for now can this be renamed to showcase that it's statefulset specific? Something like `ASSERT-STATEFULSET-VOLUME-MOUNTS-VERIFICATION`.

We could recursively crawl the members of a struct looking for PodTemplates using `reflect` 😅

Reply:
I changed the annotation. I will leave enhancing to find PodTemplates using `reflect` for later :)